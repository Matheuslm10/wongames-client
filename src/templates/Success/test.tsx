import 'session.mock'
import 'match-media-mock'
import { render, screen } from 'utils/test-utils'

import Success from '.'

import gamesMock from 'components/GameCardSlider/mock'
import highlightMock from 'components/Highlight/mock'

const props = {
  recommendedTitle: 'You may like these games',
  recommendedGames: gamesMock,
  recommendedHighlight: highlightMock
}

jest.mock('hooks/use-cart', () => ({
  ...jest.requireActual('hooks/use-cart'),
  clearCart: jest.fn()
}))

jest.mock('templates/Base', () => ({
  __esModule: true,
  default: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Mock Base">{children}</div>
  }
}))

jest.mock('components/Showcase', () => ({
  __esModule: true,
  default: function Mock() {
    return <div data-testid="Mock Showcase" />
  }
}))

describe('<Success />', () => {
  it('should render correctly', () => {
    const { container } = render(<Success {...props} />)

    expect(
      screen.getByRole('heading', { name: /Your purchase was successful!/i })
    ).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
