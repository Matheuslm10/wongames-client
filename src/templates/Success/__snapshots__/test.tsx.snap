// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Success /> should render correctly 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  width: 100%;
  max-width: 130rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(3.2rem / 2);
  padding-right: calc(3.2rem / 2);
}

.c1 {
  padding: 5.6rem 0 calc(5.6rem * 3);
}

.c2 {
  color: #FAFAFA;
  font-size: 2.8rem;
  text-align: center;
}

.c3 {
  text-align: center;
  padding: 3.2rem;
}

.c3 svg {
  color: #FAFAFA;
  background: #F231A5;
  border-radius: 50%;
  padding: 1rem;
  width: 7rem;
}

.c5 {
  color: #FAFAFA;
  font-size: 1.6rem;
  text-align: center;
  max-width: 60rem;
  margin: auto;
}

.c5 a {
  color: #F231A5;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  data-testid="Mock Base"
>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Your purchase was successful!
      </h1>
      <div
        class="c3"
      >
        <svg
          aria-hidden="true"
          class="c4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
      </div>
      <p
        class="c5"
      >
        Wait for your payment details by email. Your game is now available for download inside your
         
        <a
          href="/profile/orders"
        >
          Orders List
        </a>
        . Enjoy!
      </p>
    </div>
  </div>
  <div
    data-testid="Mock Showcase"
  />
</div>
`
